ISPN-13050 Gustavo's encoding doc review (#9327)
infinispan/infinispan: documentation/src/main/asciidoc/topics/con_cache_encoding.adoc

== Remote caches

{brandname} Server stores entries in remote caches with the encoding that is set in the cache configuration.

Hot Rod and REST clients include a media type with each request they make to {brandname} Server.
To handle multiple clients making read and write requests with different media types, {brandname} Server converts data on-demand to and from the media type that is set in the cache configuration.

> If the remote cache does not have any encoding configuration, {brandname} Server stores keys and values as generic `byte[]` without any media type information.
> In this case {brandname} Server returns keys and values for any client requests as they are stored in the remote cache, without any media type conversion.

Reviewer: ```suggestion
If the remote cache does not have any encoding configuration, {brandname} Server stores keys and values as generic `byte[]` without any media type information, that can lead to unexpected results when converting data to clients asking for different formats
```

== Remote caches

{brandname} Server stores entries in remote caches with the encoding that is set in the cache configuration.

Hot Rod and REST clients include a media type with each request they make to {brandname} Server.
To handle multiple clients making read and write requests with different media types, {brandname} Server converts data on-demand to and from the media type that is set in the cache configuration.

If the remote cache does not have any encoding configuration, {brandname} Server stores keys and values as generic `byte[]` without any media type information.
In this case {brandname} Server returns keys and values for any client requests as they are stored in the remote cache, without any media type conversion.

.Use ProtoStream encoding

> {brandname} Server cannot convert data to and from certain media types and generates an error when client requests include a media type that is not supported.

Reviewer 2: I would focus on request+cache media type pairs, because whether the server can process a request or not depends on both. E.g.

```suggestion
{brandname} Server returns an error when the client request includes a media type that cannot be converted to or from the cache encoding's media type.
```
